test/traversals/all_simple_paths.jl

```diff
@@ -0,0 +1,124 @@
+@testset "All simple paths" begin
+    # single path
+    g = path_graph(4)
+    paths = all_simple_paths(g, 1, 4)
+    @test Set(paths) == Set([[1, 2, 3, 4]])
+    @test Set(collect(paths)) == Set([[1, 2, 3, 4]])
```

Comment:
why the double test every time?

Reply:
Just a result of copying the old implementation (that had special-cased the `collect` thing). Found two instance of it; kept one to ensure that `collect` keeps working as intended.